server/game/cards/05-LoCR/CerseiLannister.js

```diff
@@ -8,22 +9,20 @@ class CerseiLannister extends DrawCard {
         });
         this.reaction({
             when: {
-                'onCardDiscarded:aggregate': (event) =>
-                    event.events.some(
-                        (discardEvent) =>
-                            discardEvent.cardStateWhenDiscarded.controller !== this.controller &&
-                            discardEvent.cardStateWhenDiscarded.location === 'hand'
-                    ) && this.allowGameAction('gainPower')
+                onCardDiscarded: {
+                    aggregateBy: (event) => ({
+                        controller: event.cardStateWhenDiscarded.controller,
+                        location: event.cardStateWhenDiscarded.location
+                    }),
+                    condition: (aggregate, events, context) =>
+                        aggregate.controller !== this.controller &&
+                        aggregate.location === 'hand' &&
+                        !context.player.isSupporter(aggregate.controller)
+                }
```

Comment:
This system seems super flexible, I'm just wondering if there are ways to make it read better. Maybe simplifying the aggregateBy in some way (pre-made functions, less reliance on `cardStateWhenXed`)

Reply:
@ystros have updated this to be a bit more simple, with just providing a map of named properties. They are they accessible through `context.aggregate.{property}`